### DISCUSSION TITLE
Project config, variables and parameters

### DISCUSSION THREAD

**Discussion by Tanushree2904** [diffplug/blowdryer#36] [Q&A] [answered]
Hello Team,

I am exploring blowdryer plugin and have a few queries regarding project configurations.
1. Is there a way I can pass any variable from build.gradle(of my project) to any config file (e.g. maven.gradle) to apply/remove some settings? 
2. Can I parameterise the plugin version in "plugin.versions" file and store them in gradle.properties?

**Comment by nedtwigg** [comment]
> Is there a way I can pass any variable from build.gradle(of my project) to any config file (e.g. maven.gradle) to apply/remove some settings?

Yes, that is possible, and a good example of that is here:

https://github.com/diffplug/atplug/blob/4108c68ac750ba33968407b46ead009b476d2d61/build.gradle#L34-L50

> Can I parameterise the plugin version in "plugin.versions" file and store them in gradle.properties?

Possible, but a bit tricky. The API for `plugin.versions` is this

https://github.com/diffplug/blowdryer/blob/c6c8056cd614eb4463f037429e35a2a2e49606f0/src/main/java/com/diffplug/blowdryer/BlowdryerSetup.java#L425-L448

An example of it in use is

https://github.com/diffplug/goomph/blob/a4b7eee3099b75d9e941033750e007e432f9186f/settings.gradle#L23-L31

You could use `replace` to replace tokens from the `plugin.versions` file, but I believe `gradle.properties` is not present when `settings.gradle` is evaluated. You would have to explicitly parse `gradle.properties` yourself. That can be done, and an example of that is here

https://github.com/mytakedotorg/mtdo/blob/e1e9bd41744d33b4481fd7398b16c85ea1b98cf3/buildSrc/build.gradle#L16-L26

**Comment by Tanushree2904** [comment]
In the below link , you are applying different settings to different sub projects.

https://github.com/diffplug/atplug/blob/4108c68ac750ba33968407b46ead009b476d2d61/build.gradle#L34-L50

I want to know if we can achieve the below scenario with 2 different projects:

Project 1:
Project 1 uses the below maven publish block in build.gradle :
```
plugins {
    id 'maven-publish'
}
java {
    withSourcesJar()
    withJavadocJar()
}

publishing {
    publications {
        //maven publishing content
    }
}
```

Project 2:
Project 2 uses only the maven publish plugin and publishing block:

```
plugins {
    id 'maven-publish'
}

publishing {
    publications {
        //maven publishing content
    }
}
```

I want to configure my maven.gradle in this way:

```
plugins {
    id 'maven-publish'
}

if javadocJarNeeded== True {    // some condition like this. (javadocJarNeeded will be passed from build.gradle of project1 )
      java {
         withSourcesJar()
         withJavadocJar()
   }
}

publishing {
    publications {
        //maven publishing content
    }
}
```

Can achieve this scenario if Project 1 and project 2 are not related? Or, is this plugin only for multi-module projects which are related like in the example you shared?

**Reply by nedtwigg** [reply, marked as answer]
> Can achieve this scenario if Project 1 and project 2 are not related?

Yes. Like this:

```gradle
// in build.gradle
ext.javadocJarNeeded = 'true'
ext.javadocJarNeeded = 'false'
apply from: 干.file('base/maven.gradle')

// in base/maven.gradle
def javadocJarNeeded = Boolean.parseBoolean(干.proj('javadocJarNeeded', 'true or false'))
```

**Comment by Tanushree2904** [comment]
Thank you so much !
